src/parse.rs

```diff
@@ -143,6 +144,24 @@ pub struct Parser<'input, 'callback> {
     link_stack: LinkStack,
 }
 
+impl<'input, 'callback> std::fmt::Debug for Parser<'input, 'callback> {
```

Comment:
Note that users of the library cannot access many of the internals of the parser that you're exposing here, so I'm not sure how useful adding them is. I'd limit the output to fields that have a public type. That also cuts down on the number of debug implementations we have to add, reducing library size (a little).

Reply:
I've added the commit to stop exposing the fields that have private types in Debug implementation